src/overlay/soil.py

```diff
@@ -450,3 +444,60 @@ def determine_tile_type(self, pos):
         if all((hoed_left, hoed_right, hoed_below)) and not hoed_above:
             return "lrt"
         return "o"
+
+
+class SoilManager:
+    all_sprites: AllSprites
+    frames: dict
+
+    _areas: dict[StudyGroup, SoilArea | None]
+
+    def __init__(self, all_sprites: AllSprites, frames: dict):
+        self.all_sprites = all_sprites
+        self.frames = frames
+
+        self._areas = {i: SoilArea(self.all_sprites, self.frames) for i in StudyGroup}
+
+    def get_area(self, study_group: StudyGroup) -> SoilArea:
+        area = self._areas[study_group]
+        return area
+
+    def load_area(
+        self,
+        study_group: StudyGroup,
+        layer: TiledTileLayer,
+        previous_soil_data: dict | None = None,
+    ):
+        self.get_area(study_group).create_soil_tiles(
+            layer, previous_soil_data=previous_soil_data
+        )
+
+    def all_soil_sprites(self):
+        groups = []
+        for area in self._areas.values():
+            groups.append(area.soil_sprites)
+        return groups
```

Comment:
It doesn't seem this function is used outside of updating the save file with new soil data, so you should turn this into a generator, or just outright return `self._areas.values()` to be more efficient

Reply:
You're right! Nice catch 😅